modal/_container_entrypoint.py

```diff
@@ -671,14 +688,16 @@ def run_inputs(input_id: str, function_call_id: str, args: Any, kwargs: Any) ->
                 run_inputs(input_id, function_call_id, args, kwargs)
     finally:
         if imp_fun.obj is not None and not imp_fun.is_auto_snapshot:
-            logger.debug("Running cls exit methods (sync)")
-            exit_methods: Dict[str, Callable] = _find_callables_for_obj(imp_fun.obj, _PartialFunctionFlags.EXIT)
-            for exit_method in exit_methods.values():
-                with function_io_manager.handle_user_exception():
-                    # We are deprecating parameterized exit methods but want to gracefully handle old code.
-                    # We can remove this once the deprecation in the actual @exit decorator is enforced.
-                    args = (None, None, None) if method_has_params(exit_method) else ()
-                    exit_method(*args)
+            # Ignore termination signals from the worker while running container @exit() methods.
+            with ignore_sigint():
```

Comment:
The closing of this context manager re-enables the signal handler, but do we need to do that? After the exit handler is done, the rest is just Modal code to exit the program right?

Reply:
Probably not — SIGINT shouldn't interrupt volume commits. We can refactor it some more here!

Reply:
Done! I ended up having to restore the signal handler at the end regardless in order for `container_test.py` to work due to some hygiene, but I added a comment explaining what's going on.